packages/flutter_tools/lib/src/hot.dart

```diff
@@ -178,26 +178,38 @@ class HotRunner extends ResidentRunner {
       return 2;
     }
 
-    if (observatoryPortCompleter != null && result.hasObservatory)
-      observatoryPortCompleter.complete(result.observatoryPort);
-
     await connectToServiceProtocol(result.observatoryPort);
 
     if (device.needsDevFS) {
+      try {
+        Uri baseUri = await _initDevFS();
+        if (connectionInfoCompleter != null) {
+          connectionInfoCompleter.complete(
+            new DebugConnectionInfo(result.observatoryPort, baseUri: baseUri.toString())
+          );
+        }
+      } catch (error) {
+        printError('Error initializing DevFS: $error');
+        return 3;
+      }
       _loaderShowMessage('Connecting...', progress: 0);
-      bool result = await _updateDevFS(
+      bool devfsResult = await _updateDevFS(
         progressReporter: (int progress, int max) {
-          _loaderShowMessage('Syncing files to device...', progress: progress, max: max);
+          if (progress % 10 == 0)
```

Comment:
This dials back the number of updates by an order to magnitude. When starting up something like flutter_gallery, we initially sync ~1,800 files, which sends 3x that number of messages (the loader message, progress value, and max). I see a 0.8 - 1.0 second speedup when launching against the ios simulator if we send fewer messages.

We could also change to doing something like only sending a message if a value has changed, or only updating the progress value every 100ms or so.